client/service/src/lib.rs

```diff
@@ -332,6 +345,35 @@ async fn build_network_future<
 	}
 }
 
+/// Set storage `NODE_ALLOWLIST` means it's a permissioned network,
+/// then only connect to these well known peers.
+fn check_node_allowlist<
+	B: BlockT,
+	BE: BackendT<B>,
+	C: BlockchainEvents<B> + StorageProvider<B, BE> + HeaderBackend<B>,
+	H: sc_network::ExHashT
+> (
+	network: &sc_network::NetworkWorker<B, H>,
+	client: &Arc<C>,
+) {
+	let id = BlockId::hash(client.info().best_hash);
+	let allowlist_storage = client.storage(&id, &StorageKey(well_known_keys::NODE_ALLOWLIST.to_vec()));
+	if let Ok(Some(raw_allowlist)) = allowlist_storage {
+		let node_allowlist: Result<Vec<NodePublic>, _> = Decode::decode(&mut &raw_allowlist.0[..]);
```

Comment:
```suggestion
		let node_allowlist: Vec<NodePublic>::decode_all(&mut &raw_allowlist.0[..]);
```

You want to use `decode_all`, not just `decode`.

Reply:
Manual fixed.